extensions/dwdm/dwdm.yml

```diff
@@ -2,18 +2,220 @@
+
 nodes:
-  - name: DWDMChannel
+  - name: DwdmDevice
+    label: DWDM Device
+    description: A DWDM device for managing optical network traffic.
+    icon: mdi:transit-connection-variant
+    namespace: Dcim
+    human_friendly_id:
+      - name__value
+    order_by:
+      - name__value
+    display_labels:
+      - name__value
+    inherit_from:
+      - DcimPhysicalDevice
+    attributes:
+      - name: name
+        kind: Text
+        unique: true # TODO: maybe combine location / name
+        order_weight: 1000
+      - name: description
+        kind: Text
+        optional: true
+        order_weight: 2000
+    relationships:
+      - name: front_interfaces
+        peer: DcimFrontDwdmInterface
+        identifier: front_interfaces
+        optional: true
+        cardinality: many
+        kind: Component
+      - name: rear_interface
+        peer: DcimRearDwdmInterface
+        identifier: rear_interfaces
+        optional: true
+        cardinality: one
+        kind: Component
+
+  - name: FrontDwdmInterface
+    label: DWDM Front Interfaces
+    menu_placement: DcimGenericDwdmInterface
+    namespace: Dcim
+    human_friendly_id:
+      - dwdm_device__name__value
+      - name__value
+    display_labels:
+      #  - dwdm  # FIXME: DcimInterface.display_labels: this property only supports attributes, not relationships
+      - name__value
+    order_by:
+      - name__value
+    uniqueness_constraints:
+      - [dwdm_device, name__value]
+    inherit_from:
+      - DcimEndpoint
+      - DcimGenericDwdmInterface
+    relationships:
+      - name: dwdm_device
+        peer: DcimDwdmDevice
+        order_weight: 900
+        identifier: front_interfaces
+        optional: false
+        cardinality: one
+        kind: Parent
+      - name: channels
+        peer: DcimDwdmChannel
+        order_weight: 1200
+        optional: true
+        cardinality: many
+        kind: Attribute
+
+  - name: RearDwdmInterface
+    label: DWDM Rear Interfaces
+    menu_placement: DcimGenericDwdmInterface
+    namespace: Dcim
+    human_friendly_id:
+      - dwdm_device__name__value
+      - name__value
+    display_labels:
+      #  - dwdm  # FIXME: DcimInterface.display_labels: this property only supports attributes, not relationships
+      - name__value
+    order_by:
+      - dwdm_device__name__value
+      - name__value
+    uniqueness_constraints:
+      - [dwdm_device, name__value]
+    inherit_from:
+      - DcimEndpoint
+      - DcimGenericDwdmInterface
+    relationships:
+      - name: dwdm_device
+        peer: DcimDwdmDevice
+        order_weight: 900
+        identifier: rear_interface
+        optional: false
+        cardinality: one
+        kind: Parent
+
+  - name: DwdmChannel
     namespace: Dcim
     description: A DWDM channel with its wavelength and frequency.
     label: DWDM Channel
     icon: game-icons:laser-warning
+    menu_placement: DcimDwdmDevice
```

Comment:
I would put those "DWDM Device" under Device instead and let the DwdmChannel by itself or in Other